### PR TITLE
New compose box design

### PR DESCRIPTION
**Pull request by kunall17**
This is currently just for design suggestions and not for code review, after the correct design is finalized, I'll complete this! 

**First default screen**

**Stream**

**Private**

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@kunall17, it looks like you've already signed the Dropbox CLA. Thanks!

**Comment by borisyankov** [comment]
I think that part of the UI is one of the weak points of the web app, and is made much worse by the nature of mobile - I really didn't like the usability in the Android app.

Typing is much harder, and way more error prone on mobile.
We should not require a person to know another person's email so they can write to them. I am not even sure why we would want to support the 'write to another person, while looking at different narrow' workflow.

Maybe we should just defer that to an in-person discussion with the other guys. It is such a weird concept, that if needed can be implemented better.

**Comment by borisyankov** [comment]
Also, we will have another line under the ComposeText (that is hidden right now, will be shown soon), and we have this always pinned subhead in the messages, and when the keyboard is popped, if we add another line we have almost no space for the messages.

What we definitely are missing is creating a new topic, but this concept needs work.

**Comment by kunall17** [comment]
> We should not require a person to know another person's email so they can write to them.

Agreed, I'll make sure in the final version its based on the Names and not the emails

>  'write to another person, while looking at different narrow' 

It often happens with me that I'm reading a stream and I need to message some other stream/person, it comes handy at that time. 
So I kinda like this UI. 
To make things easier whenever a narrow is made we can make it default that it's gonna send to that narrow! 

Moreover this UI does not flashes all the time that you'll be sending messages to this group/private which looked kinda awkward in native android, hence looks better. 
Also on tapping a message we can fill in that message recipient details, which is convenient to use!

**Comment by timabbott** [comment]
> It often happens with me that I'm reading a stream and I need to message some other stream/person, it comes handy at that time.

Yeah, this is something that we hard a lot from users; it's nice to be able to refer to some messages in a stream when PMing someone (e.g. to non-publicly point their attention at a public conversation).

I think maybe what we should do is have the default compose state be a super light experience that just sends to the current narrow, but have a button that expands compose to a full version, where you can specify where you want to send to as a new messages, whether private or stream with a new or existing topic, etc.

**Comment by borisyankov** [comment]
That makes sense; compared to the web version you are likely looking at just a few lines.

Creating a new topic could be a custom case - it can be one of the buttons next to 'image', 'photo', 'video' and when clicked it might replace them with input temporarily.

**Comment by kunall17** [comment]
@borisyankov So I should move the bubble (mode switcher from the 1st image) button to the ComposeOptions row and let the ComposeText be un touched?

**Comment by borisyankov** [comment]
@kunall17 so, we are splitting this task into two. First one, we start immediately. Next one will be 'soon but not next'.

Immediately:
* if the user is typing in a stream narrow, show a button to change the topic

Not that soon:
* fully custom 'compose screen' where you can pick stream, topic, recipients etc.

**Comment by kunall17** [comment]
Will start work from now, for the design modifications I pushed the mode changer button above 

With options 


On option change


Without divider: 


Also what to go for, with or without long divider (Message box and compose options divider)?

**Comment by kunall17** [comment]
@borisyankov 
Updated, one thing is left which is on narrow changed the inputs should change or when mode is changed it should pick up the current narrow, i'll do it soon! 

**Screencast:** 
ezgif-3-81c231c925

**Comment by kunall17** [comment]
@borisyankov Updated the branch, lets keep these feature in this PR and other work in another PR 

Added night theme support as well
screen shot 2017-05-14 at 5 28 22 pm

**Comment by kunall17** [comment]
Updated :)

**Comment by borisyankov** [comment]
High-level changes needed:
Remove the stream picker and leave the `Topic` edit only.
Comment out the extra compose buttons.

**Comment by kunall17** [comment]
Updated the PR, though it does not uses the `lastTopic` you pushed recently.
Up for a review! :)

**Comment by kunall17** [comment]
@saketkumar95 

screen shot 2017-06-02 at 12 24 24 pm 1

It looked fine on this emulator, can you specify more?

**Comment by saketkumar95** [comment]
Just forgot, It's Android 7.1 and i got this on personal device.

**Comment by saketkumar95** [comment]
@borisyankov @kunall17 We really need to get this merged soon. My work uploading images/files will be done after this is merged.

File: src/i18n/translations/messages_en.json [hunk 1]
```diff
@@ -47,5 +47,7 @@
   "No Internet connection": "No Internet connection",
   "Settings": "Settings",
   "Night mode": "Night mode",
-  "Language": "Language"
+  "Language": "Language",
+  "Topic": "Topic",
+  "Type a message here": "Type a message here"
```
Comment (saketkumar95) [inline src/i18n/translations/messages_en.json]: @kunall17 I think these should be added to all the other translations too?

Comment (borisyankov) [inline src/i18n/translations/messages_en.json]: Good that you noticed, @saketkumar95 
I added them for him with my last commit, so now it is OK.

**Comment by borisyankov** [comment]
Yeah, @kunall17 can you update the PR today?

**Comment by borisyankov** [comment]
The historical PR, merged at last :D

**Comment by kunall17** [comment]
Haha, finally :)

**Pull request merged**